xen/arch/arm/coproc/coproc.c

```diff
@@ -353,24 +351,27 @@ struct coproc_device *coproc_alloc(struct platform_device *pdev,
 
     for ( i = 0; i < num_mmios; ++i )
     {
-        res = platform_get_resource(pdev, IORESOURCE_MEM, i);
-        coproc->mmios[i].base = devm_ioremap_resource(dev, res);
-        if ( IS_ERR(coproc->mmios[i].base) )
+        struct mmio *mmio = &coproc->mmios[i];
+        ret = dt_device_get_address(np, i, &mmio->addr, &mmio->size);
+        if ( ret )
         {
-            printk("Failed to remap IO for \"%s\"\n", dev_path(coproc->dev));
-            ret = PTR_ERR(coproc->mmios[i].base);
+            printk("Failed to get mmio index %d for \"%s\"\n",
```

Comment:
Code with this prints is being refactored, so I would not focus on prints changes. Otherwise all similar prints should be reworded in the code.